thoth/python/source.py

```diff
@@ -40,9 +40,10 @@
 from thoth.common.helpers import parse_datetime
 
 _LOGGER = logging.getLogger(__name__)
+LEGACY_URLS = {"https://pypi.python.org/simple": "https://pypi.org/simple"}
 
 
-@attr.s(frozen=True, slots=True)
+@attr.s(frozen=False, slots=True)
```

Comment:
This could introduce issues on adviser side that expects source to be hashable. Any reason why it's not frozen anymore?

Reply:
Seems like I can't use the `validator` decorator with the frozen property set to `true`, any suggestions?

Reply:
OK, I can try to test the code with adviser to see if this will work. :+1:

Reply:
BTW what about using callable validators?

https://www.attrs.org/en/stable/init.html#callables

Reply:
Even with a callable validator, the frozen needs to be False to change the value of an attribute.

Reply:
```
@dataclass(frozen=True)
class Source:
    value: InitVar[str]
    url = attrib.ib(type=str, init=False)

    def __post_init__(self, value: str):
        url = LEGACY_URLS.get(value) or value
```

Could you try something like this?
https://docs.python.org/3/library/dataclasses.html#init-only-variables

Reply:
Do we want to switch to `dataclasses`?